contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,504 @@
+
+There is two kinds of conflict to handle in policies of the same priority level: Structural and semantical conflicts.
+
+
+Structural conflicts must handled at creation time as much as possible, on the other hand, semantical conflicts should be handled at runtime: detect that there's a conflict and emit an event stating that policies couldn't be satisfied due to a conflict.
+
+
+### How policies are computed
+
+Policies may have overlapping rules, to handle this policies are computed in the following order:
+
+- compute policies at `exception` priority.
+- compute policies at `cluster` priority.
+- compute policies at `user` priority.
+- compute policies at `default` priority.
+
+If a policy doesn't specify a priority the default priority applies.
+
+they should also obey to the following rules:
+
+- everything that is required is by definition allowed.
+
+
+# Detailed Design
+
+
+## SchedulingPolicy
+
+Proposed API group: `policy/v1alpha1`
+
+
+### SchedulingPolicy content
+
+SchedulingPolicy spec is composed of optional fields that allow scheduling rules. If a field is absent from a SchedulingPolicy, this `SchedulingPolicy` won't allow any item from the missing field.
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  priority: <exception,cluster,user,default>
+  namespaces:
+    matchNames:
+      - ns1
+      - ns2
+    matchLabels:
+      key: value
+  action:  <required,allowed,denied>
+  rules:
+    schedulerNames:
+       match: []
+    priorityClassNames:
+       match: []
+    tolerations:
+       match: []
+    nodeSelectors:
+       match: []
+    nodeAffinities:
+       match: []
+    podAntiAffinities:
+       match: []
+    podAffinities:
+       match: []
+```
+
+
+### required
+Elements here are required, pods won't schedule if they aren't present. Also note that if something is required it is also allowed.
+
+### allowed
+Elements here are allowed, the policy allows the presence of these elements. From Pod's perspective, a pod can use one or N of the allowed items.
+
+### deny
+If pods specify one of these, the pod won't schedule to a node, we won't dive into deny as it is the exact opposite of required.
+
+### Scheduler name
+
+If `schedulerNames` is absent from `allowed`, `default` or `required`, no scheduler is allowed by this specific policy.
+
+#### required
+
+Require that pods use either the green-scheduler (which is used by specifying `spec.schedulerName` in pod definition) or the `my-scheduler` scheduler (by specifying `spec.schedulerName: "my-scheduler"`):
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  namespaces:
+    matchNames:
+      - default
+  action: required
+  rules:
+    schedulerNames:
+       match: ["green-scheduler","my-scheduler"]
+```
+
+An empty list of schedulerNames has no effect, as pod use the `defaultScheduler` if no scheduler is specified:
```

Comment:
We should take a consistent approach towards `unset` vs `empty` rules. In my opinion a `required` rules with an `empty` field, means that the pod must leave the field empty. In your example, the pod is required to leave the `schedulerName` as empty. An `unset` field is different. An unset field means that the pod does not have any restrictions for the field.